Add Obsidian vault for docs rewrite

hakusaro opened the pull request:
Because Readme.io has been kinda scummy in terms of "giving us the
features that we want" and because most wikis don't work well, I'm
proposing a hybrid approach: an Obsidian vault (which is plain markdown)
and having the contents published to Obsidian publish. The end result
will be nice a nice and easily navigable website that can be used to
reference TShock things without needing a separate interface to edit the
docs for.

Plus we can integrate with CI, etc.

Major things to do:

- [ ] Undo the changes to the dumper utilities so that plain markdown tables are generated
- [ ] Ideally supply things like CI to alert if things need to be updated (changed permissions file? maybe you should possibly have re-run the dumper and updated the permissions docs?)
- [ ] Get Obsidian publish and set it up
- [ ] Manually transfer over pages worth keeping
- [ ] Re-verify the protocol docs

hakusaro committed:
Add Obsidian vault for docs rewrite

Because Readme.io has been kinda scummy in terms of "giving us the
features that we want" and because most wikis don't work well, I'm
proposing a hybrid approach: an Obsidian vault (which is plain markdown)
and having the contents published to Obsidian publish. The end result
will be nice a nice and easily navigable website that can be used to
reference TShock things without needing a separate interface to edit the
docs for.

Plus we can integrate with CI, etc.

hakusaro committed:
∆: make gitignore nicer

drunderscore committed:
Docs: Begin refactoring of developer protocol documentation.

This is the beginning of a complete rewrite of the "Multiplayer Packet
Structure" page, with more Obsidian-ism to make traversing nice.

This is only the first refactor so hopefully we'll be able to come up
with a common scheme and layout soon.

Something I just noticed that I have missed is the direction in which
these packets are sent (Client -> Server, Server -> Client, or Server
<-> Client)

drunderscore committed:
Docs: Insert intended network direction at the top of packets.

drunderscore committed:
Docs: Reformat Player Info and Sync Inventory Slot packets

drunderscore committed:
Docs: Document more packets and some new, never properly before
documented ones! Explain what tile sections are.

drunderscore committed:
Docs: Document even more packets!

drunderscore commented:
Looks like you added `docs/.obsidian/*` to the git ignore, but added and committed `docs/.obsidian/config` before then.

hakusaro commented:
@drunderscore  yes, intentionally

hakusaro committed:
Merge remote-tracking branch 'origin/general-devel' into h/docs

hakusaro committed:
∆: modify markdown output for permissions to add wikilinks

hakusaro committed:
Merge branch 'h/docs' of github.com:Pryaxis/TShock into h/docs

hakusaro committed:
∆: add more docs

hakusaro committed:
Merge branch 'h/docs' of github.com:Pryaxis/TShock into h/docs

hakusaro committed:
∆: more docs

hakusaro committed:
∆: more dosc

hakusaro committed:
∆: document very important stuff

hakusaro committed:
∆: more docs

hakusaro committed:
∆: more docs

hakusaro committed:
Merge pull request #2526 from drunderscore/docs/protocol

Begin documentation of Terraria network protocol

hakusaro committed:
WHY ARE THERE SO MANY COMMANDS IN THIS MOD

hakusaro committed:
Update changelog

hakusaro committed:
Merge branch 'general-devel' into h/docs

hakusaro committed:
Merge remote-tracking branch 'origin/general-devel' into h/docs

drunderscore committed:
Document that all data types are little-endian ordered

drunderscore committed:
Don't include the packet ID or direction in the heading itself

This allows anchors to each pages #ID and #Direction to be identical, as
suggested by Durtle.

drunderscore committed:
Prefer footings rather than the description for some docs

drunderscore committed:
Prefer a lesser extreme heading for non-primary parts of docs

Also, we should always end footers with a period

drunderscore committed:
Add and document missing field on Set User Slot

drunderscore committed:
Document packets that require server side characters to be enabled

drunderscore committed:
Correct formatting

drunderscore committed:
Document all the things

drunderscore committed:
Document Set Time and Toggle Door

drunderscore committed:
Document Sync Tile Rect

drunderscore committed:
Document Sync Item (and Sync Instanced Item)

drunderscore committed:
Document lots more packets!

Arthri committed:
Turn packet list into packet table

hakusaro committed:
Merge pull request #2537 from drunderscore/docs/protocol

Document more Terraria Protocol

Arthri committed:
Fix NPC being spaced

Arthri committed:
Undefined isn't suppose to be page yet

hakusaro committed:
Merge pull request #2538 from Arthri/docs/protocol_arthri

Turn packet list into table
